04-Intro_to_Vue/assignment/css/_settings.responsive.scss

```diff
@@ -0,0 +1,52 @@
+// Media Query Generator
+
+@mixin media-query($mq) {
+
+    $breakpoint-found: false;
+
+    // Loop through the list of breakpoints we’ve provided in our settings file.
+    @each $breakpoint in $breakpoints {
+
+        // Grab the alias and the condition from their respective locations in
+        // the list.
+        $alias:     nth($breakpoint, 1);
+        $condition: nth($breakpoint, 2);
+
+        // If the media query we’ve specified has an alias and a condition...
+        @if $mq == $alias and $condition {
+
+            // ...tell the mixin that we’ve found it...
+            $breakpoint-found: true;
+
+            // ...and spit it out here.
+            @media #{$condition} {
+                @content;
+            }
+
+        }
+
+    }
+
+    // If the user specifies a non-exitent alias, send them a warning.
+    @if $breakpoint-found == false{
+        @warn "Oops! Breakpoint ‘#{$mq}’ does not exist."
+    }
+
+}
+
+// Add breakpoints
+$breakpoints: (
+    "xl"            "screen and (min-width: 1300px)",
+    "desk"          "screen and (min-width: 1024px)",
+    "lg"            "screen and (min-width: 920px)",
+    "tab"           "screen and (min-width: 768px)",
+    "mobile"        "screen and (max-width: 767px)",
```

Comment:
mixing min-width and max width might give you inconsistent results. I'd keep it to one or the other.

Reply:
This is super helpful!